components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -0,0 +1,776 @@
+		players[1] = {
+			displayname = name,
+			name = TBD:upper(),
+			flag = Flags.CountryName(flag),
+			extradata = {faction = Faction.read(race) or Faction.defaultFaction}
+		}
+		extradata.hasRaceOrFlag = true
+	end
+
+	return {
+		type = opponent.type,
+		name = name,
+		score = opponent.score,
+		extradata = extradata,
+		match2players = players
+	}
+end
+
+---@param opponent table
+---@param partySize integer
+---@return table
+function CustomMatchGroupInput.processPartyOpponentInput(opponent, partySize)
+	local players = {}
+	local links = {}
+
+	for playerIndex = 1, partySize do
+		local name = Logic.emptyOr(opponent['p' .. playerIndex], opponent[playerIndex]) or ''
+		local link = mw.ext.TeamLiquidIntegration.resolve_redirect(Logic.emptyOr(
+				opponent['p' .. playerIndex .. 'link'],
+				Variables.varDefault(name .. '_page'),
+				name
+			)):gsub(' ', '_')
+		table.insert(links, link)
+
+		table.insert(players, {
+			displayname = name,
+			name = link,
+			flag = Flags.CountryName(Logic.emptyOr(
+					opponent['p' .. playerIndex .. 'flag'],
+					Variables.varDefault(name .. '_flag')
+				)),
+			extradata = {faction = Faction.read(Logic.emptyOr(
+					opponent['p' .. playerIndex .. 'race'],
+					Variables.varDefault(name .. '_race')
+				)) or Faction.defaultFaction}
+		})
+	end
+
+	table.sort(links)
+
+	return {
+		type = opponent.type,
+		name = table.concat(links, ' / '),
+		score = opponent.score,
+		extradata = opponent.extradata,
+		match2players = players
+	}
+end
+
+---@param opponent table
+---@param date string
+---@return table
+function CustomMatchGroupInput.ProcessTeamOpponentInput(opponent, date)
+	local template = string.lower(Logic.emptyOr(opponent.template, opponent[1], '')--[[@as string]]):gsub('_', ' ')
+
+	if String.isEmpty(template) or template == 'noteam' then
+		return Opponent.blank(Opponent.team)
+	end
+
+	if not mw.ext.TeamTemplate.teamexists(template) then
+		return opponent
+	end
+
+	local templateData = mw.ext.TeamTemplate.raw(template, date)
+
+	opponent.icon = templateData.image
+	opponent.icondark = Logic.emptyOr(templateData.imagedark, templateData.image)
+	opponent.name = templateData.page:gsub(' ', '_')
+	opponent.template = templateData.templatename or template
+
+	return opponent
+end
+
+--MapInput functions
+
+---@param match table
+---@param mapIndex integer
+---@param subGroupIndex integer
+---@return integer
+function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+	local map = Json.parseIfString(match['map' .. mapIndex])
+	map.map = (MapsData[(map.map or ''):lower()] or {}).name or map.map
+
+	-- set initial extradata for maps
+	map.extradata = {
+		comment = map.comment or '',
+		header = map.header or '',
+	}
+
+	-- inherit stuff from match data
+	map.type = match.type
+	map.liquipediatier = match.liquipediatier
+	map.liquipediatiertype = match.liquipediatiertype
+	map.game = match.game
+	map.date = map.date or match.date
+	map.patch = match.patch
+
```

Comment:
This can use the parent stuff that CS uses?

Reply:
not all of it but most